### PR TITLE
chore(eng-prod): Enable datadog events on release [INFRA-53]

### PR DESCRIPTION
**Pull request by Justinon**
Task: https://staging.clickup.com/t/333/INFRA-256

- chore: Convert `.projenrc.js` to `.projenrc.ts` so we can import TS modules into the project AND reveal problems with the configuration itself
- fix: Correct the placement of invalid key `semanticTitleOptions` to start actually enforcing scope requirement
- chore: Enable DD event sending on release

### CODE REVIEW

File: .projenrc.ts [hunk 1]
```diff
@@ -110,4 +115,5 @@ new YamlFile(project, 'codecov.yml', {
   },
 });
 
+datadog.addReleaseEvent(project);
```
Comment (ahammond) [inline .projenrc.ts:118]: This is a library. Why are we eventing releases for it? I can see eventing when a consumer changes the version of the library they're using, but... that's completely separate. Unless the goal is to track compliance of projects in keeping their version of `clickup-projen` up-to-date. But... that seems like a pretty lofty goal for some day in the future, not now.

**Review by ahammond — APPROVED** [review]
Love the move to `.projenrc.ts`.

File: .projenrc.ts (hunk 1 above)
Comment (Justinon) [reply .projenrc.ts:118]: I...think that's fair actually upon more thought. I'll remove the emitting of the event on release and keep the move to `.projenrc.ts`.

**Comment by Justinon** [comment]
Closed in favor of https://github.com/time-loop/clickup-projen/pull/63

**Pull request closed without merging**